bin/cylc-diff

```diff
@@ -128,9 +128,8 @@ def get_option_parser():
 
 @cli_function(get_option_parser)
 def main(parser, options, *args):
-    srv_files_manager = SuiteSrvFilesManager()
-    suite1, suite1rc = srv_files_manager.parse_suite_arg(options, args[0])
-    suite2, suite2rc = srv_files_manager.parse_suite_arg(options, args[1])
+    suite1, suite1rc = suite_srv_files_mgr.parse_suite_arg(options, args[0])
+    suite2, suite2rc = suite_srv_files_mgr.parse_suite_arg(options, args[1])
```

Comment:
Just import `parse_suite_arg`, to be consistent with what you did in `bin/cylc-check-versions`?

Reply:
So there are two reasons I diddn't do this:

1) I'm kinda lazy sometimes and this way its a quick and simple search n replace (my bad).
2) We have unit tests which work by hacking the output of `suite_srv_files_mgr` functions. As far as I can see for mock patching to work we need to import the module not its functions.

I'll import specific functions where possible (e.g. this example) but we will still have a few `suite_srv_files_mgr` imports after.

Reply:
>As far as I can see for mock patching to work we need to import the module not its functions.

I think we should be able to find a workaround. `test_scan.py` mocks functions in a module... it does not mock the imported functions. These imported functions will exist in the test class scope. But if you need to mock the functions in the imported module, that's still possible.

i.e., these two functions imported in [`test_scan.py`](https://github.com/cylc/cylc-flow/blob/e2582e4b2ae1e3ada5b3bb0c9b4657c6fdc2c798/cylc/flow/tests/network/test_scan.py#L26) are under test. So we are calling them directly.

These functions will call other functions from the `cylc.flow.network.scan` module. I **think** (haven't tested) that I could even mock `get_scan_items_from_fs` or `re_compile_filters`. But they cannot be mocked in the test scope. They would have to be mocked as `patch('cylc.flow.network.scan.re_compile_filters')`...

Not really good explaining it, sorry. But if needed I can try to help with this later.

Reply:
Thanks.